internal/server/iigointernal/executive.go

```diff
@@ -0,0 +1,148 @@
+package iigointernal
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/voting"
+)
+
+type executive struct {
+	ID               shared.ClientID
+	clientPresident  roles.President
+	budget           shared.Resources
+	speakerSalary    shared.Resources
+	RulesProposals   []string
+	ResourceRequests map[shared.ClientID]shared.Resources
+}
+
+// returnSpeakerSalary returns the salary to the common pool.
+func (e *executive) returnSpeakerSalary() shared.Resources {
+	x := e.speakerSalary
+	e.speakerSalary = 0
+	return x
+}
+
+// Get rule proposals to be voted on from remaining islands
+// Called by orchestration
+func (e *executive) setRuleProposals(rulesProposals []string) {
+	e.RulesProposals = rulesProposals
+}
+
+// Set approved resources request for all the remaining islands
+// Called by orchestration
+func (e *executive) setAllocationRequest(resourceRequests map[shared.ClientID]shared.Resources) {
+	e.ResourceRequests = resourceRequests
+}
+
+// Get rules to be voted on to Speaker
+// Called by orchestration at the end of the turn
+func (e *executive) getRuleForSpeaker() string {
+	e.budget -= serviceCharge
+	result, _ := e.clientPresident.PickRuleToVote(e.RulesProposals)
+	return result
+}
+
+// Send Tax map all the remaining islands
+// Called by orchestration at the end of the turn
+func (e *executive) getTaxMap(islandsResources map[shared.ClientID]shared.Resources) map[shared.ClientID]shared.Resources {
+	e.budget -= serviceCharge
+	result, _ := e.clientPresident.SetTaxationAmount(islandsResources)
```

Comment:
#127

Reply:
If I am not mistaken here returning an error is needed because that's also how we will handle islands choosing to not perform IIGO actions. In the MVP islands didn't have this choice. What should happen here is getTaxMap also returning an error and the calling function broadcastTaxation not executing (and hence not deducting resources from the budget)

Reply:
Yeah I think @Spruudis is right here, but like suggested in some of the comments on this PR we will move these error usages into booleans